src/Runner.Worker/ActionManager.cs

```diff
@@ -355,6 +396,26 @@ public Definition LoadAction(IExecutionContext executionContext, Pipelines.Actio
                         Trace.Verbose($"Details: {StringUtil.ConvertToJson(compositeAction?.Steps)}");
                         Trace.Info($"Load: {compositeAction.Outputs?.Count ?? 0} number of outputs");
                         Trace.Info($"Details: {StringUtil.ConvertToJson(compositeAction?.Outputs)}");
+                        if (CachedChildPreSteps.TryGetValue(action.Id, out var preSteps))
+                        {
+                            compositeAction.PreSteps = preSteps;
```

Comment:
wondering whether we should change so we cache all (only argument is consistency)